langchain/llms/clarifai.py

```diff
@@ -13,92 +13,107 @@
                 user_id=USER_ID, app_id=APP_ID, model_id=MODEL_ID)
     """
 
-    stub: Any  #: :meta private:
-    metadata: Any
-    userDataObject: Any
+  stub: Any  #: :meta private:
+  userDataObject: Any
 
-    model_id: Optional[str] = None
-    """Model id to use."""
+  model_id: Optional[str] = None
+  """Model id to use."""
 
-    model_version_id: Optional[str] = None
-    """Model version id to use."""
+  model_version_id: Optional[str] = None
+  """Model version id to use."""
 
-    app_id: Optional[str] = None
-    """Clarifai application id to use."""
+  app_id: Optional[str] = None
+  """Clarifai application id to use."""
 
-    user_id: Optional[str] = None
-    """Clarifai user id to use."""
+  user_id: Optional[str] = None
+  """Clarifai user id to use."""
 
-    clarifai_pat_key: Optional[str] = None
+  pat: Optional[str] = None
 
-    api_base: str = "https://api.clarifai.com"
+  api_base: str = "https://api.clarifai.com"
 
-    stop: Optional[List[str]] = None
+  class Config:
+    """Configuration for this pydantic object."""
 
-    class Config:
-        """Configuration for this pydantic object."""
+    extra = Extra.forbid
 
-        extra = Extra.forbid
-
-    @root_validator()
-    def validate_environment(cls, values: Dict) -> Dict:
-        """Validate that we have all required info to access Clarifai
+  @root_validator()
+  def validate_environment(cls, values: Dict) -> Dict:
+    """Validate that we have all required info to access Clarifai
         platform and python package exists in environment."""
-        values["clarifai_pat_key"] = get_from_dict_or_env(
-            values, "clarifai_pat_key", "CLARIFAI_PAT_KEY"
-        )
-        user_id = values.get("user_id")
-        app_id = values.get("app_id")
-        model_id = values.get("model_id")
-
-        if values["clarifai_pat_key"] is None:
-            raise ValueError("Please provide a clarifai_pat_key.")
-        if user_id is None:
-            raise ValueError("Please provide a user_id.")
-        if app_id is None:
-            raise ValueError("Please provide a app_id.")
-        if model_id is None:
-            raise ValueError("Please provide a model_id.")
-        return values
-
-    @property
-    def _default_params(self) -> Dict[str, Any]:
-        """Get the default parameters for calling Cohere API."""
-        return {}
-
-    @property
-    def _identifying_params(self) -> Dict[str, Any]:
-        """Get the identifying parameters."""
-        return {**{"model_id": self.model_id}}
-
-    @property
-    def _llm_type(self) -> str:
-        """Return type of llm."""
-        return "clarifai"
-
-    def _call(
-        self,
-        prompt: str,
-        stop: Optional[List[str]] = None,
-        run_manager: Optional[CallbackManagerForLLMRun] = None,
-        **kwargs: Any
-    ) -> str:
-        """Call out to Clarfai's PostModelOutputs endpoint.
+    values["pat"] = get_from_dict_or_env(values, "pat", "CLARIFAI_PAT")
```

Comment:
this is a breaking change, are we sure we want to make it?

Reply:
yeah, the previous one wasn't aligned with how our python client normally takes in the PAT like the ClarifaiAuthHelper class used within here, so wanted to clean that up while the integration is still new.